### PR TITLE
Make the logger be concurrenctly safe to use

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

Try to fix tikv/pd#3159.

### CODE REVIEW

**Comment by JmPotato** [comment]
@nolouch @lysu PTAL

**Review by lysu — COMMENTED** [review]
LGTM

**Comment by nolouch** [comment]
lgtm

**Comment by HunDunDM** [comment]
The race is caused by the test code, but the PR seems to degrade the code performance, please pay attention.
Now `log.Debug` also needs an atomic.

File: log.go [hunk 1]
```diff
@@ -86,29 +98,24 @@ func newStdLogger() (*zap.Logger, *ZapProperties) {
...
 // ReplaceGlobals replaces the global Logger and SugaredLogger.
-// It's unsafe for concurrent use.
+// It's safe for concurrent use.
 func ReplaceGlobals(logger *zap.Logger, props *ZapProperties) {
-	_globalL = logger
-	_globalS = logger.Sugar()
-	_globalP = props
+	_globalL.Store(logger)
+	_globalS.Store(logger.Sugar())
+	_globalP.Store(props)
```
Comment (xczh) [inline log.go:118]: 这看起来不是个正确的实现。逻辑上来说，L、S和P三者应该是一致的（同时更改），`ReplaceGlobals`通过三个原子操作无法保证这一点 : )

This doesn't seem to be the right implementation. Logically speaking, L, S and P should be consistent (changed at the same time), `ReplaceGlobals` cannot guarantee this through three atomic operations.

**Pull request merged**